steps.go

```diff
@@ -107,7 +107,9 @@ func (tc *TestContext) InitializeScenario(ctx *godog.ScenarioContext) {
 	tempDir := os.TempDir()
 	tc.serviceDiscoveryDirectory = path.Join(tempDir, "service-discovery")
 	if _, err := os.Stat(tc.serviceDiscoveryDirectory); err == nil {
-		_ = os.RemoveAll(tc.serviceDiscoveryDirectory)
+		if err = os.RemoveAll(tc.serviceDiscoveryDirectory); err != nil {
+			tc.logger.Warningf("failed to remove previous service directory data directory %s (%v)", tc.serviceDiscoveryDirectory, err)
+		}
 	}
 	if err := os.Mkdir(tc.serviceDiscoveryDirectory, os.ModePerm); err != nil {
 		tc.logger.Warningf("failed to create service discovery data directory %s (%v)", tc.serviceDiscoveryDirectory, err)
```

Comment:
### Reason

- At times the `tempDir` could not be removed (Windows 10, Version 2004 (Build 19041.572), Docker Engine v19.03.13)
- Logging a warning at removal, might further improve error tracing (no error is returned, if directory does not exist)

### Further improvements

- `os.MkdirAll ` could be used instead of `os.Mkdir`, which does **not** return an error if the directory already exists
(thereby logging more specific warnings, where they really occur (at removal vs creation)